x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/risk_score_data_client.ts

```diff
@@ -39,6 +42,7 @@ interface RiskScoringDataClientOpts {
 }
 
 export class RiskScoreDataClient {
+  private alreadyUpgraded = false;
```

Comment:
I have one more bit of validation I need to do, which is to ensure that the task manager always uses a single `RiskScoreDataClient` instance. I'll repost here if I find that it does not, which might necessitate a small change to this code.

Reply:
Found that the Kibana task manager does indeed create separate data client instances for each iteration of the task. Also found that the task itself actually creates parallel calls to the `calculate_and_persist` function, which was not properly handled. (_Also_ found that all this required a bit more than a "small change to this code" 😅 )

[This commit](https://github.com/elastic/kibana/pull/174474/commits/a8fb56e68bbed33edaeb245f05ef73e922d24d45) hopefully alleviates these issues. I've added some unit test cases that flex those possibilities.